src/arithmetic.jl

```diff
@@ -0,0 +1,56 @@
+Base.promote_rule(::Type{Arf}, ::Type{<:Union{AbstractFloat,Integer}}) = Arf
+Base.promote_rule(::Type{Arb}, ::Type{<:Union{AbstractFloat,Integer,Arf,ArbRef}}) = Arb
+Base.promote_rule(
+    ::Type{Acb},
+    ::Type{<:Union{AbstractFloat,Integer,Complex,Arf,Arb,ArbRef,AcbRef}},
+) = Acb
```

Comment:
We could also promote `Rational` and do we have to specify which subtypes of `Complex` we want to promote?

Reply:
I aded the promotion from Rational, and restricted `Complex` to `Complex{<:Union{AbstractFloat,Integer,Rational}}`